src/ol_infrastructure/infrastructure/operations/elasticsearch.py

```diff
@@ -0,0 +1,211 @@
+        )
+    ],
+)
+
+instance_type_name = (
+    elasticsearch_config.get("instance_type") or InstanceTypes.medium.name
+)
+instance_type = InstanceTypes[instance_type_name].value
+elasticsearch_instances = []
+elasticsearch_export = {}
+subnets = destination_vpc["subnet_ids"]
+subnet_id = subnets.apply(chain)
+for count, subnet in zip(range(elasticsearch_config.get_int("instance_count") or 3), subnets):  # type: ignore # noqa: WPS221
+    instance_name = f"elasticsearch-{environment_name}-{count}"
+    salt_minion = OLSaltStackMinion(
+        f"saltstack-minion-{instance_name}",
+        OLSaltStackMinionInputs(minion_id=instance_name),
+    )
+
+    cloud_init_userdata = build_userdata(
+        instance_name=instance_name,
+        minion_keys=salt_minion,
+        minion_roles=["elasticsearch"],
+        minion_environment=environment_name,
+        salt_host=f"salt-{ops.env_suffix}.private.odl.mit.edu",
+        additional_cloud_config={
+            "device_aliases": {"ephemeral0": "/dev/nvme1n1"},
+            "disk_setup": {
+                "ephmeral0": {"table_type": "gpt", "layout": True, "overwrite": False}
+            },
+            "fs_setup": [
+                {
+                    "device": "ephemeral0",
+                    "partition": "none",
+                    "label": "None",
+                    "filesystem": "ext4",
+                }
+            ],
+            "mount": ["ephemeral0", "/var/lib/elasticsearch", "ext4"],
+        },
+    )
+
+    instance_tags = aws_config.merged_tags(
+        {"Name": instance_name, "elasticsearch_env": environment_name}
+    )
+    elasticsearch_instance = ec2.Instance(
+        f"elasticsearch-instance-{environment_name}-{count}",
+        ami=debian_10_ami.id,
+        user_data=cloud_init_userdata,
+        instance_type=instance_type,
+        iam_instance_profile=elasticsearch_instance_profile.id,
+        tags=instance_tags,
+        volume_tags=instance_tags,
+        subnet_id=subnet,
+        key_name=salt_config.require("key_name"),
+        root_block_device=ec2.InstanceRootBlockDeviceArgs(
+            volume_type="gp2",
+            volume_size=20,
+            encrypted=True,
+        ),
+        ebs_block_devices=ec2.InstanceEbsBlockDeviceArgs(
+            device_name="/dev/nvme1n1",
+            volume_type="gp2",
+            volume_size=100,
+            encrypted=True,
+        ),
+        vpc_security_group_ids=[
+            destination_vpc["security_groups"]["salt_minion"],
+            elasticsearch_security_group.id,
+        ],
+        opts=ResourceOptions(depends_on=[salt_minion]),
+    )
+    elasticsearch_instances.append(elasticsearch_instance)
+
+    elasticsearch_export[instance_name] = {
+        "public_ip": elasticsearch_instance.public_ip,
+        "private_ip": elasticsearch_instance.private_ip,
+        "ipv6_address": elasticsearch_instance.ipv6_addresses,
+    }
+
+export(
```

Comment:
Exports should be delegated to `__main__.py`, so this should be set as a dictionary with a variable name of `export_data`.

Reply:
The security groups should be set as a dictionary assigned to a variable name of `security_groups`